src/main/java/xyz/nkomarn/harbor/provider/DefaultAFKProvider.java

```diff
@@ -0,0 +1,85 @@
+package xyz.nkomarn.harbor.provider;
+
+import org.bukkit.Bukkit;
+import org.bukkit.configuration.ConfigurationSection;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.Listener;
+import org.bukkit.event.player.PlayerQuitEvent;
+import org.jetbrains.annotations.NotNull;
+import xyz.nkomarn.harbor.Harbor;
+import xyz.nkomarn.harbor.api.AFKProvider;
+import xyz.nkomarn.harbor.listener.AfkListeners;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+public class DefaultAFKProvider implements AFKProvider, Listener {
+    private final Harbor harbor;
+    private final boolean enabled;
+    private final Map<UUID, Instant> playerActivity;
+    private final AfkListeners listeners;
+
+    public DefaultAFKProvider(@NotNull Harbor harbor) {
+        this.harbor = harbor;
+        playerActivity = new HashMap<>();
+        if (enabled = (harbor.getConfig().getBoolean("afk-detection.fallback-enabled", true))) {
+            harbor.getLogger().info("Registering fallback AFK detection system.");
+            listeners = new AfkListeners(this);
+            harbor.getServer().getPluginManager().registerEvents(this, harbor);
```

Comment:
`listeners` needs to be registered not `this`.

Also:
Better call `enableListeners()` and move the print and registration to this method.
Move the object creation to enable listener as well. See below for reasons.

Reply:
Okay this makes sense now with the stuff above... I think this whole thing should be changed...

There is now need to seperate the default AFK provider from the AFK listener. This just causes confusion.